libminifi/src/capi/api.cpp

```diff
@@ -57,38 +57,28 @@ class DirectoryConfiguration {
   }
 };
 
-nifi_port *create_port(const char *port) {
-  if (nullptr == port)
-    return nullptr;
-  nifi_port *p = new nifi_port();
-  p->port_id = new char[strlen(port) + 1];
-  memset(p->port_id, 0x00, strlen(port) + 1);
-  strncpy(p->port_id, port, strlen(port));
-  return p;
-}
-
-int free_port(nifi_port *port) {
-  if (port == nullptr)
-    return -1;
-  delete[] port->port_id;
-  delete port;
-  return 0;
-}
-
 /**
  * Creates a NiFi Instance from the url and output port.
  * @param url http URL for NiFi instance
  * @param port Remote output port.
+ * @Deprecated for 0.6.0 in favor of
```

Comment:
Sorry to nitpick, but shall we refer to minifi releases or API version, could see this getting a little confusing at some point?  How could we best document these version relationships?

Reply:
Oh that's a very good point.

Reply:
Well rereading the deprecation note, it makes more sense than my first take on things.  I guess I anticipate this maybe being problematic in the near future when both would reach the 1.0 mark.  I know there is connectedness between the both the capi and the agent itself but would almost prefer just saying that it was done for 0.02.  I could see the API reaching a certain level of maturity where the agent would continue to evolve but the API itself is fairly static.  

As a general thought, we could probably use some exploration of minifi cpp's versioning/support guarantees especially as this API gets built out.